src/mongocxx/uri.cpp

```diff
@@ -129,5 +130,160 @@ class write_concern uri::write_concern() const {
         stdx::make_unique<write_concern::impl>(libmongoc::write_concern_copy(wc)));
 }
 
+static stdx::optional<stdx::string_view> _string_option(mongoc_uri_t* uri, std::string opt_name) {
+    const char* value;
+
+    value = libmongoc::uri_get_option_as_utf8(uri, opt_name.c_str(), NULL);
+    if (!value) {
+        return {};
+    }
+
+    return stdx::string_view{value};
+}
+
+static stdx::optional<std::int32_t> _int32_option(mongoc_uri_t* uri, std::string opt_name) {
```

Comment:
libmongoc provides a `mongoc_uri_get_option_as_int32`, but I chose not to use it.

Like `mongoc_uri_get_option_as_utf8` it accepts a fallback argument to return if the option is unset. But it treats an option in a URI set to 0 as unset (e.g. `mongodb://host/?connectTimeoutMS=0`). But some options allow 0 (e.g. `localThresholdMS`, `connectTimeoutMS`).

http://mongoc.org/libmongoc/current/mongoc_uri_get_option_as_int32.html